src/cpp/src/whisper/models/statefull_decoder.hpp

```diff
@@ -12,7 +12,8 @@ class WhisperStatefullDecoder : public WhisperDecoder {
 public:
     WhisperStatefullDecoder(const std::filesystem::path& models_path,
                             const std::string& device,
-                            const ov::AnyMap& properties);
+                            const ov::AnyMap& properties,
+                            const ov::PartialShape& lhs_shape);
```

Comment:
can we obtain the required dimensions from decoder model? From its parameters or results? As I understand, you need for updating KV cache. Also, you can use variables of decoder for this, can't you?
My point is to avoid passing last hidden state shape from encoder. I think we can deduce it from decoder itself.